modules/core/misc/java/test/MatTest.java

```diff
@@ -455,6 +455,20 @@ public void testGetIntIntShortArray() {
         bytesNum = sm.get(1, 1, buff11);
         assertEquals(4, bytesNum);
         assertTrue(Arrays.equals(new short[] {340, 341, 0, 0}, buff11));
+
+        Mat m2 = new Mat(new int[]{ 5, 5, 5 }, CvType.CV_16S);
+        short[] data = new short[125];
+        for (int i = 0; i < data.length; i++ ) {
+            data[i] = (short)i;
+        }
+        m2.put(new int[] {0, 0, 0}, data);
+        Mat matNonContinuous = m2.submat(new Range[]{new Range(1,4), new Range(1,4), new Range(1,4)});
+        Mat matContinuous = matNonContinuous.clone();
+        short[] outNonContinuous = new short[27];
+        matNonContinuous.get(new int[] { 0, 0, 0 }, outNonContinuous);
+        short[] outContinuous = new short[27];
+        matContinuous.get(new int[] { 0, 0, 0 }, outContinuous);
+        assertArrayEquals(outNonContinuous, outContinuous);
```

Comment:
`assertMatEqual` does not check continuity flag. You can try it instead of custom method.

Reply:
@asmorkalov not sure what you mean. The bug this PR is fixing is in the output from the `get` function which is an array. So we need to compare the arrays not the Mats.

Reply:
You are right. Sorry, missed `get` call in the code.